theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -316,7 +317,22 @@ export class TraceViewerWidget extends ReactWidget {
                 await this.fetchAnnotationCategories(output);
                 this.update();
             } else {
-                document.getElementById(exist.id)?.scrollIntoView();
+                const traceId = this.openedExperiment.UUID;
+                if (document.getElementById(traceId + exist.id + 'focusContainer')) {
+                    document.getElementById(traceId + exist.id + 'focusContainer')?.focus();
```

Comment:
This doesn't work for the datatree, it outputs this warning to console:

> root WARN Widget was activated, but did not accept focus after 2000ms

It appears that datatree has no component that accepts focus so focus() cannot be used to make it scroll into view?

Also, my intention was to use setFocus() method for an output component to implement to decide which of its children should receive focus. But here we do not use this method, instead the id suffix 'focusContainer' is used to indicate which one it should be. It seems we should not have two ways to achieve the same goal? Maybe scrolling into view and assigning focus should be two different steps?

Reply:
@PatrickTasse updated datatree-output-component and rebased to resolve conflicts